matchzoo/contrib/models/esim.py

```diff
@@ -0,0 +1,241 @@
+        >>> model.params['mlp_activation_func'] = 'tanh'
+        >>> model.params['mask_value'] = 0
+        >>> model.params['dropout_rate'] = 0.5
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """Get default parameters."""
+        params = super().get_default_params(with_embedding=True,
+                                            with_multi_layer_perceptron=True)
+
+        params.add(Param(
+            name='dropout_rate',
+            value=0.5,
+            desc="The dropout rate for all fully-connected layer"
+        ))
+
+        params.add(Param(
+            name='lstm_dim',
+            value=300,
+            desc="The dimension of LSTM layer."
+        ))
+
+        params.add(Param(
+            name='mask_value',
+            value=0,
+            desc="The value would be regarded as pad"
+        ))
+
+        return params
+
+    def _make_inputs(self) -> list:
+        input_left = keras.layers.Input(
+            name='text_left',
+            shape=self._params['input_shapes'][0]
+        )
+        input_right = keras.layers.Input(
+            name='text_right',
+            shape=self._params['input_shapes'][1]
+        )
+        return [input_left, input_right]
+
+    def _make_embedding_layer(self, name='embedding') -> keras.layers.Layer:
+        """
+        Overwrite the BaseModel._make_embedding_layer to allow
+        specifiying mask_zero.
+
+        :param name: name for embedding layer, default 'embedding'
+        """
+        return keras.layers.Embedding(
+            self._params['embedding_input_dim'],
+            self._params['embedding_output_dim'],
+            trainable=self._params['embedding_trainable'],
+            mask_zero=True,
+            name=name
+        )
+
+    def _expand_dim(self, inp, axis) -> keras.layers.Layer:
+        """
+        Wrap keras.backend.expand_dims into a Lambda layer.
+
+        :param inp: input tensor to expand the dimension
+        :param axis: the axis of new dimension
+        """
+        return keras.layers.Lambda(lambda x: K.expand_dims(x, axis=axis))(inp)
+
+    def _make_atten_mask_layer(self) -> keras.layers.Layer:
+        """
+        Make mask layer for attention weight matrix so that
+        each word won't pay attention to <PAD> timestep.
+        """
+        return keras.layers.Lambda(
+            lambda weight_mask: weight_mask[0] + (1.0 - weight_mask[1]) * -1e7,
+            name="atten_mask")
+
+    def _make_BiLSTM_layer(self, lstm_dim) -> keras.layers.Layer:
+        """
+        Bidirectional LSTM layer in ESIM.
+
+        :param lstm_dim: int, dimension of LSTM layer
+        :return: `keras.layers.Layer`.
+        """
+        return keras.layers.Bidirectional(
+            layer=keras.layers.LSTM(lstm_dim, return_sequences=True),
+            merge_mode='concat')
+
+    def _max(self, texts, mask) -> tf.Tensor:
```

Comment:
be consistent and annotate `text` and `mask` with type hints.

Or delete all type hints for internal methods.